### ISSUE TITLE
java.lang.NoSuchMethodException: sun.security.ssl.SSLSocketImpl.getHost() on jdk8u352-b08

### ISSUE THREAD

**Issue by gullux** [adoptium/adoptium-support#711]
Executing code
**_final Method getHostMethod = socket.getClass().getDeclaredMethod("getHost");_**

No exception on Oracle JDK jdk1.8.0_311

Should the getHost method be deprecated?

**Comment by gullux** [comment]
Same issue on [jdk8u362-b09]

**Comment by jerboaa** [comment]
`sun.security.ssl.SSLSocketImpl` is an internal class and should not be used directly. It risks seing breakages like you see by calling into it reflectively. From that standpoint, I'm not sure there is anything to fix in upstream JDK 8 unless there is a good reason to. I'd strongly suggest to find a way to only use *API* [methods](https://docs.oracle.com/javase/8/docs/api/javax/net/ssl/SSLSocket.html) which won't break next time you upgrade your JDK.

What are you trying to do? Why are you using `getHost()`?

**Comment by gullux** [comment]
This is old code and we would migrate our environments to Temurin JDK.
I don't understand why remove a method instead of deprecate it. In this way we can suppose that Temurin JDK is not 100% compliant with the Java Specs.
If the method will be not included in the next release, we will decide if rewrite the code or use another JDK distro.

Another segnalation https://youtrack.jetbrains.com/issue/WI-45045

**Comment by jerboaa** [comment]
> I don't understand why remove a method instead of deprecate it.

Going back to `jdk8u262` I've found the method. It's gone in OpenJDK 8u since the `8u272` update. That release included the TLS 1.3 support. Note that the method was **package-private** in package `sun.security.ssl`. Here it is:

```
    //
    // ONLY used by ClientHandshaker for the server hostname during handshaking
    //
    synchronized String getHost() {
        // Note that the host may be null or empty for localhost.
        if (host == null || host.length() == 0) {
            useImplicitHost(true);
        }

        return host;
    }
```

Why not deprecate it? Because it was an internal **package-private** method. Not every single method in the JDK is considered public API, let alone has anything to do with the Java SE specs. I'm sorry, this broke something for you.

> In this way we can suppose that Temurin JDK is not 100% compliant with the Java Specs.

I'd argue it **is**, as it's passing the JDK 8 SE JCK for every release that's produced by the Adoptium WG.

> If the method will be not included in the next release, we will decide if rewrite the code or use another JDK distro.

It's unlikely the method will come back, to be honest. By using a different JDK distro, there is no guarantee the method will be there, though. It might be there for one release of distro X and gone in the next release of distro X. Again, because it's internal API. So you will be betting on your luck.

**Comment by jerboaa** [comment]
Also note, that this method is gone in JDK 11+ too so will prevent you from upgrading your app to a newer JDK version.
